quartz/plugins/transformers/lastmod.ts

```diff
@@ -36,23 +58,24 @@ export const CreatedModifiedDate: QuartzTransformerPlugin<Partial<Options>> = (u
         () => {
           let repo: Repository | undefined = undefined
           return async (_tree, file) => {
-            let created: MaybeDate = undefined
-            let modified: MaybeDate = undefined
-            let published: MaybeDate = undefined
+            let created: DateTime | undefined = undefined
+            let modified: DateTime | undefined = undefined
+            let published: DateTime | undefined = undefined
 
             const fp = file.data.filePath!
             const fullFp = path.isAbsolute(fp) ? fp : path.posix.join(file.cwd, fp)
             for (const source of opts.priority) {
               if (source === "filesystem") {
                 const st = await fs.promises.stat(fullFp)
-                created ||= st.birthtimeMs
-                modified ||= st.mtimeMs
+                // birthtime can be 0 on some filesystems, so default to the earlier of ctime/mtime
+                created ||= DateTime.fromMillis(st.birthtimeMs || Math.min(st.ctimeMs, st.mtimeMs))
+                modified ||= DateTime.fromMillis(st.mtimeMs)
               } else if (source === "frontmatter" && file.data.frontmatter) {
-                created ||= file.data.frontmatter.date as MaybeDate
-                modified ||= file.data.frontmatter.lastmod as MaybeDate
-                modified ||= file.data.frontmatter.updated as MaybeDate
-                modified ||= file.data.frontmatter["last-modified"] as MaybeDate
-                published ||= file.data.frontmatter.publishDate as MaybeDate
```

Comment:
tbh we should dealias all of the frontmatter stuff on the frontmatter transformers instead of doing it here.

Let me submit a quick PR for this.

Reply:
ideally, I think we should make these customizable, rather than making a list of predefined aliases. 

All the obsidian plugins I can find that set the created/modified date ([front-matter-timestamps](https://github.com/lighthousedino/obsidian-front-matter-timestamps), [frontmatter-modified-date](https://github.com/alangrainger/obsidian-frontmatter-modified-date), [update-time-updater](https://github.com/muratagawa/update-time-updater), [linter](https://github.com/platers/obsidian-linter)) allow customizing the property that stores the timestamp. It is quite likely a user wishes to use a non-default property name. 

Collapsing these as aliases in the frontmatter transformer isn't great, IMO, since it's hard-coding these assumptions even further "upstream". For example, I might want to use `published` for the date notes are published to the vault, but `publishDate` as the date a referent (e.g. a book/article) was published in real life. And I might want to display this information in the page on quartz, maybe using a new plugin.

Reply:
I think we don't have to, at the end of the day quartz is "opinionated", and we should just support the most populate case, which is what we have here (i believe this aligns with all the frontmatter supported by obsidian publish)

Reply:
fair enough. I'll wait for that PR to get merged and rebase on top of that